src/chainstate/stacks/db/blocks.rs

```diff
@@ -5002,6 +5003,10 @@ impl StacksChainState {
             None,
         )?;
 
+        let block_limit = clarity_tx
+            .block_limit()
+            .unwrap_or(ExecutionCost::max_value());
```

Comment:
hmm... what would it mean if `block_limit()` failed?

Reply:
It would mean that a transaction was currently being evaluated. I don't think it's possible in this code path, but it'd be good to warn --

```suggestion
            .unwrap_or_else(|| {
                 warn!("Failed to read transaction block limit");
                 ExecutionCost::max_value()
            });
```